pygmt/src/velo.py

```diff
@@ -238,8 +238,12 @@ def velo(self, data=None, **kwargs):
     """
     kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
```

Comment:
This isn't really related to the PR, but it's something that I found confusing while working on part 2. Why is kwargs passed to `figure._preprocess`? It doesn't look like it is used at all.

Reply:
Good question, this ensures that `velo` is plotted to the right instance of the figure (e.g. if someone does `fig1`, `fig2`, etc). See also https://github.com/GenericMappingTools/pygmt/pull/1072#discussion_r611030789.

Reply:
It makes sense to me why we call the figure method to ensure the correct figure is activated in the session. I still don't really understand why we pass and return kwargs to/from `_preprocess`, or even why we call `_preprocess` at all rather than calling `_activate_figure` directly. I pushed a [new branch](https://github.com/GenericMappingTools/pygmt/tree/preprocess-to-activate) to demonstrate the point of confusion, in which I replaced all the ``kwargs = self._preprocess(**kwargs)`` with ``self._activate_figure`` which seems more readable, a tiny bit faster, and still seems to work fine.

Reply:
This was added in #148, I'm not sure exactly why, but maybe @leouieda was thinking that there might be some other common function that needs to be called every time a plotting function happens.

Reply:
This was setup for any other preprocessing of kwargs that may be needed or other figure plotting methods. At the time, I put this in place to allow things like passing `pygmt.Figure(projection=pygmt.Mercator(), width="25c")` and then inserting the projection code into the first plot command. 

It could also be useful for any bookkeeping operation that is needed, like activating the figure, activating a subplot, etc.